magma/lib/magma/tsv_writer.rb

```diff
@@ -17,5 +30,40 @@ def write_tsv
         @payload.reset(@model)
       end
     end
+
+    def transpose_records
+      # Collect all data, then yield the transposed results
+      [].tap do |buffer|
+        buffer << tsv_row_to_array(@payload.tsv_header)
+
+        @retrieval.each_page do |records|
+          @payload.add_records(@model, records)
+          buffer.concat(tsv_records_to_array(@payload.to_tsv))
+          @payload.reset(@model)
+        end
+      end.transpose.each do |row|
+        yield array_to_tsv_row(row)
+      end
+    end
+
+    private
+
+    def tsv_row_to_array(row)
+      # Need to set `limit` for split()
+      #   to "space-pad" the resulting array
+      row = row.gsub("\n", "")
```

Comment:
Hmm, is this safe?  Any reason we arent using the `CSV` module with `sep: "\t"`?  My concern is that this kind of hand parsing won't work anytime there is an escaped string or other weird edge cases.

Reply:
Thanks, good suggestion! I'll investigate.